k8s-install/config.yaml

```diff
@@ -0,0 +1,36 @@
+# This ConfigMap can be used to configure a self-hosted Canal installation.
+# See `canal.yaml` for an example of a Canal deployment which uses
+# the config in this ConfigMap.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: canal-config 
+  namespace: kube-system
+data:
+  # Configure this with the location of your etcd cluster.
+  etcd_endpoints: "http://127.0.0.1:2379"
+
+  # The interface used by flannel for host <-> host communication.
+  flannel_iface: "eth1"
+
+  # The CNI network configuration to install on each node.  The special
+  # values in this config will be automatically populated.
+  # See here for more details: <LINK TO DOCS>
```

Comment:
I plan to add a doc in the calico-cni repo that outlines the behavior of the auto-configuration, and what the valid parameters are.